api/client/client_test.go

```diff
@@ -50,6 +51,18 @@ func TestNewServer(t *testing.T) {
 	if len(hosts) != 1 || hosts[0] != "https://1.1.1.1:8080" {
 		t.Fatalf("expected [https://1.1.1.1:8080], but have %v", hosts)
 	}
+
+	s = NewServer("https://1.1.1.1:8181/")
+	hosts = s.Hosts()
+	if len(hosts) != 1 || hosts[0] != "https://1.1.1.1:8181/" {
+		t.Fatalf("expected [https://1.1.1.1:8181/], but have %v", hosts)
+	}
+
+	s = NewServer("https://1.1.1.1/foo")
+	hosts = s.Hosts()
+	if len(hosts) != 1 || hosts[0] != "https://1.1.1.1/foo" {
+		t.Fatalf("expected [https://1.1.1.1/foo], but have %v", hosts)
+	}
```

Comment:
Please add a test for '1.1.1.1:888', '1.1.1.1/foo'.

Reply:
This revealed more shortcomings of normalizeURL who's functionality is now mostly replaced by urlx.Parse()